### ISSUE TITLE
offset in footnote numberings with latex_show_urls='footnote' and references inside footnotes

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#4850]
When document uses ``latex_show_urls='footnote'`` and some footnote contains references, the references are not converted to footnotes but the footnote numbering acquires an offset.

For example here is latex output from attached project:

```latex
automatically numbered footnote %
\begin{footnote}[1]\sphinxAtStartFootnote
This footnote contains two links:
\sphinxhref{http://wwww.sphinx-doc.org}{first link}, \sphinxhref{http://wwww.sphinx-doc.org}{second link}
%
\end{footnote}

And here is \sphinxhref{http://wwww.sphinx-doc.org}{third link}%
\begin{footnote}[4]\sphinxAtStartFootnote
\sphinxnolinkurl{http://wwww.sphinx-doc.org}
%
\end{footnote}
```

#### Expected results

Admittedly it could be a bit difficult for LaTeX writer to create footnotes from inside footnotes (not investigated) but if the links are not converted to footnotes then the footnote numbering should not acquire an offset

### Reproducible project / your project

[latexlinksasfootnotes.zip](https://github.com/sphinx-doc/sphinx/files/1912055/latexlinksasfootnotes.zip)

### Environment info
- OS: Mac
- Sphinx version: 1.7.3+/adf5e9fda
- Python version: 3.6.5 (CPython)
- Docutils version: 0.14 
- Jinja2 version: 2.10

May relate to #4827 fixing #4784, and to #4843

**Comment by tk0miya** [comment]
I changed the milestone of this to 1.8 because I just made PR #4870

**Comment by tk0miya** [comment]
Fixed by #4870.
